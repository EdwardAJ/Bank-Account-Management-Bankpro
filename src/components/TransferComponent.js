import React from 'react';
import './TransferComponent.css';

import axios from 'axios';
import { TRANSFER_URL } from './../utils/endpoints.js';
import { HEADERS } from './../utils/header.js';
import getXMLResponse from './../utils/functions.js';
import { getCookie } from './../utils/cookie.js';

export default class TransferComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            destAccount: '',
            amount: ''
        };
        this.handleAmountChange = this.handleAmountChange.bind(this);
        this.handleDestAccountChange = this.handleDestAccountChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleAmountChange (event) {
        this.setState({amount: event.target.value});
    }

    handleDestAccountChange (event) {
        this.setState({destAccount: event.target.value});
    }

    handleSubmit (event) {
        var isVirtualAccount = false;
        if (this.state.destAccount.length > 8) {
            isVirtualAccount = true;
        }
        let requestBody =
        '<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:con="http://controllers.webservice.com/">' +
            '<soapenv:Header/>' +
            '<soapenv:Body>' +
                '<con:transferFund>' +
                    '<srcAccountNumber>' + getCookie('auth') + '</srcAccountNumber>' +
                    '<destAccountNumber>' + this.state.destAccount + '</destAccountNumber>' +
                    '<amount>' + this.state.amount + '</amount>' +
                    '<isVirtualAccount>' + isVirtualAccount + '</isVirtualAccount>' +
                '</con:transferFund>' +
            '</soapenv:Body>' +
        '</soapenv:Envelope>';

        axios.post(TRANSFER_URL, requestBody, {
            headers: HEADERS
        }).then((response) => {
            if (response.status === 200) {
                let resultStr = getXMLResponse(response.data)[0].innerHTML;
                alert(resultStr);
            }
        }).catch((error) => {
            alert("Error: ", error);
        });

        event.preventDefault();
    }

    render() {
        return (
            <div className="container-margin">
                <p className="login-header"> Nomor Rekening Transfer </p>
                <div className="form-area">
                    <form>
                        <input className="form" type="text" value={ this.state.destAccount } onChange={ this.handleDestAccountChange } placeholder="Example: 13517070" />
                    </form> 
                </div>
                <p className="login-header"> Jumlah Transfer </p>
                <div className="form-area">
                    <form>
                        <input className="form" type="text" value={ this.state.amount } onChange={ this.handleAmountChange } placeholder="Example: 40000" />
                    </form> 
                </div>
                <form>
                    <button className="submit" onClick={ this.handleSubmit }> TRANSFER </button>
                </form>
            </div>
        );
    }
}