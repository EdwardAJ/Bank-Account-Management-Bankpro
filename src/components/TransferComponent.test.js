import React from 'react';
import { shallow } from 'enzyme';
import TransferComponent from './TransferComponent';

describe("TransferComponent", () => {
    it("should match this component", () => {
        const component = shallow(<TransferComponent />);
        expect(component.getElements()).toMatchSnapshot();
    });
});