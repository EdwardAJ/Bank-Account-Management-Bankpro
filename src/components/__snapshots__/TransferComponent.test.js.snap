exports[`TransferComponent should match this component 1`] = `
Array [
  <div
    className="container-margin">
    <p
        className="login-header">
         Nomor Rekening Transfer 
    </p>
    <div
        className="form-area">
        <form>
            <input
                className="form"
                onChange={[Function]}
                placeholder="Example: 13517070"
                type="text"
                value="" />
        </form>
    </div>
    <p
        className="login-header">
         Jumlah Transfer 
    </p>
    <div
        className="form-area">
        <form>
            <input
                className="form"
                onChange={[Function]}
                placeholder="Example: 40000"
                type="text"
                value="" />
        </form>
    </div>
    <form>
        <button
            className="submit"
            onClick={[Function]}>
             TRANSFER 
        </button>
    </form>
</div>,
]
`;
